error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/util/MoreJUnitJupiterMatchers.java

```diff
@@ -0,0 +1,72 @@
+package tech.picnic.errorprone.bugpatterns.util;
+
+import static com.google.errorprone.matchers.ChildMultiMatcher.MatchType.AT_LEAST_ONE;
+import static com.google.errorprone.matchers.Matchers.allOf;
+import static com.google.errorprone.matchers.Matchers.annotations;
+import static com.google.errorprone.matchers.Matchers.anyOf;
+import static com.google.errorprone.matchers.Matchers.isType;
+import static tech.picnic.errorprone.bugpatterns.util.MoreMatchers.hasMetaAnnotation;
+
+import com.google.common.collect.Iterables;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.MultiMatcher;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.AnnotationTree;
+import com.sun.source.tree.AssignmentTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.tools.javac.code.Type;
+import java.util.Optional;
+import javax.lang.model.type.TypeKind;
+
+/**
+ * A set of JUnit Jupiter-specific helper methods and {@link Matcher Matchers}, adding on to the
+ * ones from {@link com.google.errorprone.matchers.JUnitMatchers}.
+ */
+public final class MoreJUnitJupiterMatchers {
```

Comment:
I'm a bit on the fence about the naming. 
Although _within_ the class it's a good idea to make explicit it's about JUnit Jupiter things, it makes sense to me to name the class inline with the upstream one, meaning we call it `MoreJUnitMatchers`. WDYT? 

Additionally, in the near future expect more things will be added not necessarily specific to JUnit Jupiter.